File: fh6-relay/api_client.py
import aiohttp

from session_manager import LapRecord


async def submit_lap(
    api_url: str,
    token: str,
    discord_id: str,
    discord_username: str,
    lap: LapRecord,
    track: str,
    vehicle_name: str,
) -> dict:
    payload = {
        "token": token,
        "discord_id": discord_id,
        "discord_username": discord_username,
        "lap_time_ms": lap.lap_time_ms,
        "track": track,
        "vehicle_name": vehicle_name,
        "car_class_int": lap.car_class_int,
        "car_ordinal": lap.car_ordinal,
        "raw_telemetry": lap.raw_telemetry,
    }
    async with aiohttp.ClientSession() as session:
        async with session.post(f"{api_url}/api/lap", json=payload) as resp:
            data = await resp.json()
            if resp.status == 200:
                return data
            raise ValueError(data.get("reason", "unknown_error"))

File: fh6-relay/src/main/apiClient.ts
interface SubmitPayload {
  lapTimeMs: number;
  track: string;
  vehicleName: string;
  carClassInt: number;
  carOrdinal: number;
}

export class ApiClient {
  constructor(
    private readonly apiUrl: string,
    private readonly token: string,
    private readonly discordId: string,
    private readonly discordUsername: string,
  ) {}

  async getVehicles(): Promise<string[]> {
    const res = await fetch(`${this.apiUrl}/api/vehicles`);
    if (!res.ok) throw new Error(`${res.status}`);
    return res.json();
  }

  async getTracks(): Promise<string[]> {
    const res = await fetch(`${this.apiUrl}/api/tracks`);
    if (!res.ok) throw new Error(`${res.status}`);
    return res.json();
  }

  async submitLap(payload: SubmitPayload): Promise<{ entry_id: number }> {
    const res = await fetch(`${this.apiUrl}/api/lap`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        token: this.token,
        discord_id: this.discordId,
        discord_username: this.discordUsername,
        lap_time_ms: payload.lapTimeMs,
        track: payload.track,
        vehicle_name: payload.vehicleName,
        car_class_int: payload.carClassInt,
        car_ordinal: payload.carOrdinal,
      }),
    });
    if (!res.ok) throw new Error(`${res.status}`);
    return res.json();
  }
}
